lib-ee/emqx_ee_bridge/test/emqx_ee_bridge_cassa_SUITE.erl

```diff
@@ -0,0 +1,561 @@
+            ?assertMatch([#{result := {error, _}} | _], Trace),
+            [#{result := {error, Error}} | _] = Trace,
+            case Error of
+                {resource_error, _} ->
+                    ok;
+                {recoverable_error, disconnected} ->
+                    ok;
+                _ ->
+                    ct:fail("unexpected error: ~p", [Error])
+            end
+        end
+    ),
+    ok.
+
+%% This test doesn't work with batch enabled since it is not possible
+%% to set the timeout directly for batch queries
+%%
+%% XXX: parameter with request timeout is not supported yet.
+%%
+%t_write_timeout(Config) ->
+%    ProxyName = ?config(proxy_name, Config),
+%    ProxyPort = ?config(proxy_port, Config),
+%    ProxyHost = ?config(proxy_host, Config),
+%    {ok, _} = create_bridge(Config),
+%    Val = integer_to_binary(erlang:unique_integer()),
+%    SentData = #{payload => Val, timestamp => 1668602148000},
+%    Timeout = 1000,
+%    emqx_common_test_helpers:with_failure(timeout, ProxyName, ProxyHost, ProxyPort, fun() ->
+%        ?assertMatch(
+%            {error, {resource_error, #{reason := timeout}}},
+%            query_resource(Config, {send_message, SentData, [], Timeout})
+%        )
+%    end),
+%    ok.
+
+t_simple_sql_query(Config) ->
+    ?assertMatch(
+        {ok, _},
+        create_bridge(Config)
+    ),
+    Request = {query, <<"SELECT count(1) AS T FROM system.local">>},
+    Result = query_resource(Config, Request),
+    case ?config(enable_batch, Config) of
+        true -> ?assertEqual({error, {unrecoverable_error, batch_prepare_not_implemented}}, Result);
+        false -> ?assertMatch({ok, {<<"system.local">>, _, [[1]]}}, Result)
+    end,
+    ok.
+
+t_missing_data(Config) ->
+    ?assertMatch(
+        {ok, _},
+        create_bridge(Config)
+    ),
+    %% emqx_ee_connector_cassa will send missed data as a `null` atom
+    %% to ecql driver
+    Result = send_message(Config, #{}),
+    ?assertMatch(
+        %% TODO: match error msgs
+        {error, {unrecoverable_error, {8704, <<"Expected 8 or 0 byte long for date (4)">>}}},
+        Result
+    ),
+    ok.
+
+t_bad_sql_parameter(Config) ->
+    ?assertMatch(
+        {ok, _},
+        create_bridge(Config)
+    ),
+    Request = {query, <<"">>, [bad_parameter]},
+    Result = query_resource(Config, Request),
+    case ?config(enable_batch, Config) of
+        true ->
+            ?assertEqual({error, {unrecoverable_error, invalid_request}}, Result);
+        false ->
+            ?assertMatch(
+                {error, {unrecoverable_error, _}}, Result
+            )
+    end,
+    ok.
+
+t_nasty_sql_string(Config) ->
+    ?assertMatch({ok, _}, create_bridge(Config)),
+    Payload = list_to_binary(lists:seq(1, 127)),
+    Message = #{
+        topic => atom_to_binary(?FUNCTION_NAME),
+        payload => Payload,
+        timestamp => erlang:system_time(millisecond)
+    },
+    %% XXX: why ok instead of {ok, AffectedLines}?
```

Comment:
Does the driver need modification?

Reply:
Yes, can we put it in the following improvement tasks?